pkg/controller/endpoint/endpoints_controller.go

```diff
@@ -311,29 +317,19 @@ func (e *EndpointController) handleErr(err error, key interface{}) {
 	utilruntime.HandleError(err)
 }
 
-func (e *EndpointController) syncService(key string) error {
```

Comment:
Having the key here is consistent with all the other controllers.  How about just skipping the second split in this method and us ethe already split values.